planet/src/main/webapp/js/authentication.js

```diff
@@ -279,6 +280,52 @@ function signInWithProvider(provider) {
     });
 }
 
+function onForgotPassword() {
+    closeModal('sign-in-modal'); 
+    openModal('reset-pwd-modal');
+    document.getElementById('submit-message').innerHTML = '';
+    document.getElementById('email-reset-pwd').value = '';
+}
+
+function resetPassword() {
+    const email = document.getElementById('email-reset-pwd');
+    if (email === null) {
+        console.log('Email element not found');
+        return;
+    }
+    const message = document.getElementById('submit-message');
+    message.style.textAlign = 'center';
+    message.style.width = '100%';
+    message.style.height = 'auto';
+    message.style.marginBottom = '2%';
+    firebase.auth().sendPasswordResetEmail(email.value)
+    .then(function() {
+        email.value = '';
+        message.style.backgroundColor = 'lightgreen';
+        message.innerText = 'A reset email has been sent to your inbox! Check it out!';
+    })
+    .catch(function(error) {
+        message.style.backgroundColor = 'tomato';
+        const errorCode = error.code;
+        console.log(error.message);
+        if (errorCode === 'auth/invalid-email') {
+            message.innerText = 'Invalid Email! Please try again!';
+        } else if (errorCode === 'auth/missing-android-pkg-name') {
+            message.innerText = 'Reset email has not been sent! Please try again!';
+        } else if (errorCode === 'auth/missing-continue-uri') {
+            message.innerText = 'Reset email has not been sent! Please try again!';
+        } else if (errorCode === 'auth/missing-ios-bundle-id') {
+            message.innerText = 'Reset email has not been sent! Please try again!';
+        } else if (errorCode === 'auth/invalid-continue-uri') {
+            message.innerText = 'Reset email has not been sent! Please try again!';
+        } else if (errorCode === 'auth/unauthorized-continue-uri') {
+            message.innerText = 'Reset email has not been sent! Please try again!';
+        } else if (errorCode === 'auth/user-not-found') {
+            message.innerText = 'User not found! Make sure the email is correct!';
```

Comment:
This is a tricky situation. In theory, we shouldn't show this to the user because it divulges information about our set of users. 

E.g. if I'm malicious and want to know whether or not you have an account on this website, I can now find out.

Normally, we would just tell the user something like:

"If the account was found, an email was sent to user@email.com to reset the password."

Then, we would hide these errors in the server side to prevent leaking any of this info at all. However, I don't think that's possible with firebase, because this info would be present in the response object and the user can inspect it in the chrome dev tools. 

Long story short, I think this is fine given the limitations of the firebase API, but in general we would try to hide this specific error from users.

Reply:
Interesting point! So, if the email has not been sent for any reason, should I just print "If the account was found, an email was sent to user@email.com to reset the password." even though I know it hasn't been sent?

Reply:
Yes, I think that would be best here.

It doesn't actually add any real security, since the chrome dev tools will show the response, but it's the right philosophy.

Reply:
Okay I'll change that right away!